zerver/lib/avatar.py

```diff
@@ -36,6 +34,33 @@ def avatar_url(
     )
 
 
+def get_system_bots_avatar_file_name(email: str) -> str:
+    system_bot_avatar_name_map = {
+        settings.WELCOME_BOT: "welcome-bot",
+        settings.NOTIFICATION_BOT: "notification-bot",
+        settings.EMAIL_GATEWAY_BOT: "emailgateway",
+    }
+    return urljoin(STATIC_AVATARS_DIR, system_bot_avatar_name_map.get(email, "unknown"))
+
+
+def get_static_avatar_url(email: str, medium: bool) -> str:
+    avatar_file_name = get_system_bots_avatar_file_name(email)
+
+    avatar_file_name += "-medium.png" if medium else ".png"
+
+    if not DEBUG and not staticfiles_storage.exists(avatar_file_name):
+        # This logic cannot be executed while STATIC_ROOT is
+        # not defined.
+        return DEFAULT_AVATAR_FILE
+    else:
+        from django.contrib.staticfiles.finders import find
+
+        if not find(avatar_file_name):
+            raise AssertionError(f"Unknown avatar file for: {email}")
```

Comment:
This is just a net to catch problems early on if any breaking changes to how we handle system bots avatar. For example, the avatar file name got changed, they moved, etc.